### PR TITLE
Exchange don't add blocks on their own anymore

### PR DESCRIPTION
**Pull request by MichaelMure**
Follows:
- https://github.com/ipfs/go-ipfs-exchange-interface/pull/23
- https://github.com/ipfs/go-bitswap/pull/571

Note: require merging and bubbling https://github.com/ipfs/go-ipfs-exchange-interface/pull/23

### CODE REVIEW

**Comment by MichaelMure** [comment]
cc @Stebalien

**Review by Stebalien — APPROVED** [review]
LGTM (pending a merge/release of the interface library)

File: offline_test.go [hunk 1]
```diff
@@ -28,13 +28,14 @@ func TestHasBlockReturnsNil(t *testing.T) {
...
-	if err != nil {
-		t.Fail()
+	// we don't need to do that for the test, but that illustrate the normal workflow
+	if err := store.Put(context.Background(), block); err != nil {
+		t.Fatal(err)
 	}
 
-	if _, err := store.Get(context.Background(), block.Cid()); err != nil {
-		t.Fatal(err)
+	err := ex.NotifyNewBlocks(context.Background(), block)
```
Comment (Jorropo) [inline offline_test.go:36]: I'll push something that remove this test, it's literally testing that `return nil` returns `nil`, pointless.

**Review by Jorropo — APPROVED** [review]
LGTM

**Pull request merged**